src/Driver/AbstractMySQLDriver.php

```diff
@@ -130,6 +145,8 @@ private function getOracleMysqlVersionNumber(string $versionString): string
 
         if ($majorVersion === '5' && $minorVersion === '7') {
             $patchVersion ??= '9';
+        } else {
+            $patchVersion ??= '0';
```

Comment:
This fallback is added as version_compare doesn't like trailing dots.

```php
php > var_dump(version_compare('8.4.0', '8.4', '>='));
bool(true)

php > var_dump(version_compare('8.4.', '8.4', '>='));
bool(false)

php > var_dump(version_compare('8.4', '8.4', '>='));
bool(true)

php > var_dump(version_compare('8.4', '8.4.0', '>='));
bool(false)
```

Reply:
Where's the trailing dot coming from anyway?

Reply:
The line above the if-statement defines a nullable string and uses that in a concatenation in its return value.

```php
$majorVersion = $versionParts['major'];
$minorVersion = $versionParts['minor'] ?? 0;
$patchVersion = $versionParts['patch'] ?? null;

// ...

return $majorVersion . '.' . $minorVersion . '.' . $patchVersion;
```